pkg/operator/ceph/object/admin.go

```diff
@@ -136,28 +132,15 @@ func NewMultisiteContext(context *clusterd.Context, clusterInfo *cephclient.Clus
 }
 
 // GetAdminOpsEndpoint returns an endpoint that can be used to perform RGW admin ops
-// It returns an HTTPS endpoint if available. It prefers direct routes to the RGW(s).
 func GetAdminOpsEndpoint(s *cephv1.CephObjectStore) (string, error) {
 	nsName := fmt.Sprintf("%s/%s", s.Namespace, s.Name)
 
-	port, err := s.Spec.GetPort()
+	// advertise endpoint should be most likely to have a valid cert, so use it for admin ops
+	endpoint, err := s.GetAdvertiseEndpointUrl()
 	if err != nil {
-		return "", errors.Wrapf(err, "failed to get port for object store %q", nsName)
+		return "", errors.Wrapf(err, "failed to get advertise endpoint for object store %q", nsName)
 	}
-
-	domain := s.GetServiceDomainName()
-	if s.Spec.IsExternal() {
-		// if the store is external, pick a random external endpoint to use. if the endpoint is down, this
-		// reconcile may fail, but a future reconcile will eventually pick a different endpoint to try
-		endpoints := []string{}
-		for _, e := range s.Spec.Gateway.ExternalRgwEndpoints {
-			endpoints = append(endpoints, e.String())
-		}
-		idx := randSrc.Intn(len(endpoints))
```

Comment:
This block was just added a few days ago, right? So it's just refining the behavior, and nobody relying on that approach to worry about.

Reply:
It has the potential to affect users, but I thought about this a lot on Thursday and Friday. I came to the conclusion that if the first endpoint is what we advertise to OBCs (meaning OBCs can't use any of the other endpoints), then keeping the same logic for internal operations is the better thing for Rook to do. 

With OBCs, it was a problem in the past when Rook was randomly changing the OBC endpoints, so we made the OBCs standardize on the first endpoint in the list if multiple are present.

Also, COSI also only conceptualizes a single endpoint. In the wider S3 and object storage landscape, clients just use a single endpoint. Multiple endpoints just aren't really a thing. 

I don't think we can break our API by removing the list, but we can clarify what Rook does with a list.

Our recommendation for users in official docs has been to use only a single external endpoint that is a load balancer for quite a while, because of the OBC first-endpoint behavior. In reality, I don't expect this to be something that any users actually notice.

Reply:
As a follow-up, my proposal would be to continue with the change, and make sure this gets noted in upgrade docs. If we hear from users that this is important to them, we can revisit the internal behavior.